management-interface.go

```diff
@@ -215,6 +234,74 @@ func CheckInterfaceHandler(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(response)
 }
 
+// SpeakerTestHandler will show a frame from the camera to help with positioning
+func SpeakerTestHandler(w http.ResponseWriter, r *http.Request) {
+	tmpl.ExecuteTemplate(w, "speaker-test.html", nil)
+}
+
+// fileExists returns whether the given file or directory exists
+func fileExists(path string) bool {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true
+	} else {
+		return false
+	}
+}
+
+// findAudioFile locates our test audio file.  It returns true and the location of the file
+// if the file is found. And false and empty string otherwise.
+func findAudioFile(fileName string, secondaryPath string) (bool, string) {
+
+	// Check if the file is in the executable directory
+	if executablePath != "" && fileExists(executablePath+fileName) {
+		return true, executablePath
+	}
+
+	// In our default, second location?
+	if fileExists(secondaryPath + fileName) {
+		return true, secondaryPath
+	}
+
+	// Test sound not available
+	log.Printf("File " + fileName + " NOT found")
```

Comment:
It's good to keep logging limited to the upper levels where possible. It would be cleaner if this happened in the HTTP handler.